src/components/NavBar.jsx

```diff
@@ -1,17 +1,30 @@
 import { Link } from 'react-router-dom'
+import { useContext } from 'react'
+import { UserContext } from '../contexts/User.jsx'
+import { isLoggedOut } from '../utils/isLoggedOut.js'
+import { Avatar } from './Avatar.jsx'
 
 export function NavBar() {
+  const { userLogin, setUserLogin } = useContext(UserContext)
   return (
     <ul>
       <li>
-        <Link to="/">Home</Link>
+        <Link to='/'>Home</Link>
       </li>
       <li>
-        <Link to="/categories">Categories</Link>
-      </li>
-      <li>
-        <Link to="/login">Login</Link>
+        <Link to='/categories'>Categories</Link>
       </li>
+      {isLoggedOut(userLogin) ? (
+        <li>
+          <Link to='/login'>Login</Link>
+        </li>
+      ) : (
+        <li>
+          <b>{userLogin.username}</b>
+          <Avatar avatarUrl={userLogin.avatar_url}/>
+          <button onClick={() => setUserLogin({})}>Logout</button>
```

Comment:
How about displaying a message to let the user know that they need to log in first to post a comment? This would make it easier for the user 👍